woocommerce-gateway-stripe.php

```diff
@@ -544,19 +544,31 @@ public function add_emails( $email_classes ) {
 			}
 
 			/**
-			 * Register Stripe connect rest routes.
+			 * Register REST API routes.
+			 *
+			 * New endpoints/controllers can be added here.
 			 */
-			public function register_connect_routes() {
-
+			public function register_routes() {
+				/** API includes */
 				require_once WC_STRIPE_PLUGIN_PATH . '/includes/abstracts/abstract-wc-stripe-connect-rest-controller.php';
 				require_once WC_STRIPE_PLUGIN_PATH . '/includes/connect/class-wc-stripe-connect-rest-oauth-init-controller.php';
 				require_once WC_STRIPE_PLUGIN_PATH . '/includes/connect/class-wc-stripe-connect-rest-oauth-connect-controller.php';
+				require_once WC_STRIPE_PLUGIN_PATH . '/includes/admin/class-wc-rest-upe-flag-toggle-controller.php';
 
-				$oauth_init    = new WC_Stripe_Connect_REST_Oauth_Init_Controller( $this->connect, $this->api );
-				$oauth_connect = new WC_Stripe_Connect_REST_Oauth_Connect_Controller( $this->connect, $this->api );
+				$controllers = [
+					'WC_Stripe_Connect_REST_Oauth_Init_Controller' => [ $this->connect, $this->api ],
+					'WC_Stripe_Connect_REST_Oauth_Connect_Controller' => [ $this->connect, $this->api ],
+					'WC_REST_UPE_Flag_Toggle_Controller' => null,
```

Comment:
Should this register only when the `_wcstripe_feature_upe` flag (which controls the UPE preview) is enabled?

Reply:
I think it makes sense to wrap around `_wcstripe_feature_upe` flag because there is no need to load this controller otherwise.